frontend/catalyst/compiler.py

```diff
@@ -77,87 +79,196 @@ def get_executable_path(project, tool):
+        # operations into explicit sequence of operations that model the effect of this operation on the
+        # different metadata. This pass uses affine constructs to materialize these effects.
+        # Concretely, expanded-strided-metadata is used to decompose memref.subview as it has no
+        # lowering in -convert-memref-to-llvm.
+        "--expand-strided-metadata",
+        "--lower-affine",
+        "--convert-complex-to-standard",  # added for complex.exp lowering
+        "--convert-complex-to-llvm",
+        "--convert-math-to-llvm",
+        # Must be run after -convert-math-to-llvm as it marks math::powf illegal but doesn't convert it.
+        "--convert-math-to-libm",
+        "--convert-arith-to-llvm",
+        "--convert-memref-to-llvm",
+        "--convert-index-to-llvm",
+        "--convert-gradient-to-llvm",
+        "--convert-quantum-to-llvm",
+        # Remove any dead casts as the final pass expects to remove all existing casts,
+        # but only those that form a loop back to the original type.
+        "--canonicalize",
+        "--reconcile-unrealized-casts",
+    ]
+
+    @staticmethod
+    def get_output_filename(infile):
+        if not infile.endswith(".buff.mlir"):
+            raise ValueError(f"Input file ({infile}) is not a bufferized MLIR file")
+        return infile.replace(".buff.mlir", ".llvm.mlir")
+
+
+class QuantumCompilationPass(Pass):
+    """Lower gradients"""
+
+    _executable = get_executable_path("quantum", "quantum-opt")
+    _default_flags = ["--lower-gradients"]
+
+    @staticmethod
+    def get_output_filename(infile):
+        if not infile.endswith(".mlir"):
+            raise ValueError(f"Input file ({infile}) for quantum transforms is not an MLIR file")
+        return infile.replace(".mlir", ".opt.mlir")
+
+
+class LLVMDialectToLLVMIR(Pass):
+    """Convert LLVM Dialect to LLVM-IR."""
+
+    _executable = get_executable_path("llvm", "mlir-translate")
+    _default_flags = ["--mlir-to-llvmir"]
+
+    @staticmethod
+    def get_output_filename(infile):
+        if not infile.endswith(".llvm.mlir"):
+            raise ValueError(f"Input file ({infile}) is not an LLVM dialect MLIR file")
+        return infile.replace(".llvm.mlir", ".ll")
+
+
+class LLVMIRToObjectFile(Pass):
+    """LLVMIR To Object File."""
+
+    _executable = get_executable_path("llvm", "llc")
+    _default_flags = [
+        "--filetype=obj",
+        "--relocation-model=pic",
+    ]
+
+    @staticmethod
+    def get_output_filename(infile):
+        if not infile.endswith(".ll"):
+            raise ValueError(f"Input file ({infile}) for compilation is not an LLVMIR file")
+        return infile.replace(".ll", ".o")
+
+
+mlir_lib_path = get_lib_path("llvm", "MLIR_LIB_DIR")
+lrt_lib_path = get_lib_path("runtime", "RUNTIME_LIB_DIR")
+lrt_capi_path = os.path.join(lrt_lib_path, "capi")
+lrt_backend_path = os.path.join(lrt_lib_path, "backend")
```

Comment:
I thought the `l` prefix is because it is a library. Kind of like "lmlir_c_runner_utils" is also a library. Will delete at the end in case you want to take this suggestion back.

Reply:
Pretty sure it was for "lightning runtime" 😅

Reply:
https://github.com/PennyLaneAI/catalyst/pull/38/commits/0841faa0d06bc5a4b3f7d72edd5f9f4c811b4550